go/client/install_osx.go

```diff
@@ -0,0 +1,568 @@
+	if libkb.IsBrewBuild {
+		return BrewKBFSLabel.labelForRunMode(runMode)
+	}
+	return AppKBFSLabel.labelForRunMode(runMode)
+}
+
+func installKeybaseService(g *libkb.GlobalContext) error {
+	label := defaultServiceLabel(g.Env.GetRunMode())
+	binPath, err := binPath()
+	if err != nil {
+		return err
+	}
+	plistArgs := []string{"service"}
+	envVars := defaultLaunchdEnvVars(g, label)
+
+	plist := launchd.NewPlist(label, binPath, plistArgs, envVars)
+	return launchd.Install(plist, os.Stdout)
+}
+
+func uninstallKeybaseServices(runMode libkb.RunMode) {
+	launchd.Uninstall(AppServiceLabel.labelForRunMode(runMode), os.Stdout)
+	launchd.Uninstall(BrewServiceLabel.labelForRunMode(runMode), os.Stdout)
+}
+
+func installKBFSService(g *libkb.GlobalContext) error {
+	runMode := g.Env.GetRunMode()
+	label := defaultServiceLabel(runMode)
+	kbfsBinPath, err := kbfsBinPath(runMode)
+	if err != nil {
+		return err
+	}
+	mountPath := kbfsMountPath(g.Env.GetHome(), runMode)
+	plistArgs := []string{mountPath}
+	envVars := defaultLaunchdEnvVars(g, label)
+
+	plist := launchd.NewPlist(label, kbfsBinPath, plistArgs, envVars)
+	return launchd.Install(plist, os.Stdout)
+}
+
+func uninstallKBFSServices(runMode libkb.RunMode) {
+	launchd.Uninstall(AppKBFSLabel.labelForRunMode(runMode), os.Stdout)
+	launchd.Uninstall(BrewKBFSLabel.labelForRunMode(runMode), os.Stdout)
+}
+
+// Lookup the default service label for this build.
+func (l ServiceLabel) labelForRunMode(runMode libkb.RunMode) string {
+	switch runMode {
+	case libkb.DevelRunMode:
+		return fmt.Sprintf("%s.devel", l)
+	case libkb.StagingRunMode:
+		return fmt.Sprintf("%s.staging", l)
+	case libkb.ProductionRunMode:
+		return string(l)
+	default:
+		panic("Invalid run mode")
+	}
+}
+
+type InstallStatus struct {
+	Name   string          `json:"name"`
+	Status keybase1.Status `json:"status"`
+}
+
+func install(g *libkb.GlobalContext, force bool) []InstallStatus {
+	var err error
+	status := []InstallStatus{}
+
+	err = installCommandLine(g, true) // Always force CLI install
+	if err != nil {
+		status = append(status, InstallStatus{Name: "cli", Status: errorStatus("INSTALL_ERROR", err.Error())})
+	} else {
+		status = append(status, InstallStatus{Name: "cli", Status: keybase1.Status{Name: "OK"}})
+	}
+
+	err = installService(g, force)
+	if err != nil {
+		status = append(status, InstallStatus{Name: "service", Status: errorStatus("INSTALL_ERROR", err.Error())})
+	} else {
+		status = append(status, InstallStatus{Name: "service", Status: keybase1.Status{Name: "OK"}})
+	}
+
+	err = installKBFS(g, force)
+	if err != nil {
+		status = append(status, InstallStatus{Name: "kbfs", Status: errorStatus("INSTALL_ERROR", err.Error())})
+	} else {
+		status = append(status, InstallStatus{Name: "kbfs", Status: keybase1.Status{Name: "OK"}})
+	}
+
+	return status
+}
+
+func installCommandLine(g *libkb.GlobalContext, force bool) error {
+	linkPath := filepath.Join("/usr/local/bin", binName())
```

Comment:
Should this be hardcoded here? what if people need to install it somewhere else?